core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1524,6 +1524,19 @@ public static int multiply(int b0, int b1) {
     throw notArithmetic("*", b0, b1);
   }
 
+  /** Helper method for safe arithmetic functions that checks if overflow has occurred. */
+  private static boolean isOverflow(BigDecimal b0) {
+    return b0.compareTo(BigDecimal.valueOf(Long.MIN_VALUE)) < 0
+        || b0.compareTo(BigDecimal.valueOf(Long.MAX_VALUE)) > 0;
+  }
+
+  /** SQL {@code SAFE_MULTIPLY(<value0>, <value1>)} function. */
+  public static @Nullable Double safeMultiply(@PolyNull BigDecimal b0,
```

Comment:
We just have to handle BigDecimal * BigDecimal？May be we should also handle all the combination of input types?

Reply:
@zoudan I think I am going to mirror my commit from [LOG function](https://github.com/apache/calcite/commit/74bc025fbb5af4f43762926e0bec75acf6ac3025). This function has the same operand type // return type relationship so should be okay. [BigQuery docs](https://cloud.google.com/bigquery/docs/reference/standard-sql/mathematical_functions#log)